From adadf47a2d2b222ff9bf1d60f7175e44c70fa118 Mon Sep 17 00:00:00 2001
From: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Date: Fri, 8 Mar 2019 20:00:39 +0700
Subject: [PATCH 783/909] arm64: dts: r8a7795: Enable IPMMU-DS1 and IPMMU-MM

commit 993ff95764652b283eee29b25bc9bd633e9100a9 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Enable the r8a7795 device nodes for IPMMU-DS1 and the shared IPMMU-MM
device.

Signed-off-by: Hai Nguyen Pham <hai.pham.ud@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi | 8 ++++++++
 arch/arm64/boot/dts/renesas/r8a7795.dtsi     | 2 --
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi b/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
index f14579650f8a..9c160cd53cc2 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
@@ -253,6 +253,14 @@
 	gpio-ranges = <&pfc 0 32 28>;
 };
 
+&ipmmu_mm {
+	status = "disabled";
+};
+
+&ipmmu_ds1 {
+	status = "disabled";
+};
+
 &ipmmu_vi0 {
 	renesas,ipmmu-main = <&ipmmu_mm 11>;
 };
diff --git a/arch/arm64/boot/dts/renesas/r8a7795.dtsi b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
index 94ddbcbd081c..cce15399e60e 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
@@ -1444,7 +1444,6 @@
 				     <GIC_SPI 197 IRQ_TYPE_LEVEL_HIGH>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
-			status = "disabled";
 		};
 
 		ipmmu_ds0: mmu@e6740000 {
@@ -1462,7 +1461,6 @@
 			renesas,ipmmu-main = <&ipmmu_mm 1>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			#iommu-cells = <1>;
-			status = "disabled";
 		};
 
 		ipmmu_hc: mmu@e6570000 {
-- 
2.17.1

